lib/features/registration/pages/signup_page.dart

```diff
@@ -40,228 +44,257 @@ class _SignUpPageState extends State<SignUpPage> {
+                  ],
                 ),
-                SizedBox(
-                  height: size.height * 0.15,
-                ),
-              ],
+          resizeToAvoidBottomInset: false,
+          bottomSheet: stripestate.stripeStatus == StripeObjectStatus.display
+              ? _createInAppWebview(context, stripeCubit.state.stripeObject)
+              : Container(
+                  margin: const EdgeInsets.only(
+                    bottom: 30,
+                    left: 20,
+                    right: 20,
+                    top: 20,
+                  ),
+                  child: stripestate.stripeStatus == StripeObjectStatus.initial
+                      ? Column(
+                          mainAxisSize: MainAxisSize.min,
+                          children: [
+                            _buildAcceptPolicy(locals),
+                            ElevatedButton(
+                              onPressed: _isEnabled ? _register : null,
+                              style: ElevatedButton.styleFrom(
+                                disabledBackgroundColor: Colors.grey,
+                              ),
+                              child: Text(
+                                locals.next,
+                              ),
+                            )
+                          ],
+                        )
+                      : const SizedBox()),
+          body: Padding(
+            padding: const EdgeInsets.all(16),
+            child: BlocConsumer<RegistrationBloc, RegistrationState>(
+              listener: (context, state) {
+                if (state.status == RegistrationStatus.personalInfo) {
+                  context.goNamed(
+                    Pages.personalInfo.name,
+                    extra: context.read<RegistrationBloc>(),
+                  );
+                }
+
+                if (state.status == RegistrationStatus.sepaMandateExplanation) {
+                  context.goNamed(
+                    Pages.sepaMandateExplanation.name,
+                    extra: context.read<RegistrationBloc>(),
+                  );
+                }
+
+                if (state.status ==
+                    RegistrationStatus.bacsDirectDebitMandateExplanation) {
+                  context.goNamed(
+                    Pages.bacsMandateExplanation.name,
+                    extra: context.read<RegistrationBloc>(),
+                  );
+                }
+                if (state.status == RegistrationStatus.createStripeAccount) {
+                  context.read<StripeCubit>().fetchStripeCustomerCreationURLs();
+                }
+              },
+              builder: (context, state) {
+                return (state.status ==
+                            RegistrationStatus.createStripeAccount ||
+                        stripestate.stripeStatus == StripeObjectStatus.success)
+                    ? _createView(stripeCubit.state.stripeStatus)
+                    : Form(
```

Comment:
I think this build method can be splitted on smaller pieces somehow for better readability? Maybe the Form can be a separate widget or function? but it is minor

Reply:
Great point! It's done and it does help, especially because of all the conditional rendering on that page 😅